src/main/java/Dan/Parser.java

```diff
@@ -0,0 +1,57 @@
+package dan;
+
+import java.time.format.DateTimeParseException;
+
+public class Parser {
+    private TaskList tasks;
+
+    public Parser(TaskList tasks) {
+        this.tasks = tasks;
+    }
+    
+    public boolean parse(String input) {
+        String action = input.split(" ")[0];
+        try {
+            switch (action) {
+            case "bye":
+                Ui.sayonara();
```

Comment:
Sayonara isn't in English, so maybe we could change this to `Ui.bye()` instead?

Reply:
Had forgotten about this! Thanks for pointing it out